app/controllers/api/storage.php

```diff
@@ -24,9 +25,465 @@
+            ]));
+        } catch (Duplicate $th) {
+            throw new Exception('Bucket already exists', 409);
+        }
+
+        $audits
+            ->setParam('event', 'storage.buckets.create')
+            ->setParam('resource', 'storage/buckets/' . $bucket->getId())
+            ->setParam('data', $bucket->getArrayCopy())
+        ;
+
+        $usage->setParam('storage.buckets.create', 1);
+
+        $response->setStatusCode(Response::STATUS_CODE_CREATED);
+        $response->dynamic($bucket, Response::MODEL_BUCKET);
+    });
+
+App::get('/v1/storage/buckets')
+    ->desc('List buckets')
+    ->groups(['api', 'storage'])
+    ->label('scope', 'buckets.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'listBuckets')
+    ->label('sdk.description', '/docs/references/storage/list-buckets.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_BUCKET_LIST)
+    ->param('search', '', new Text(256), 'Search term to filter your list results. Max length: 256 chars.', true)
+    ->param('limit', 25, new Range(0, 100), 'Results limit value. By default will return maximum 25 results. Maximum of 100 results allowed per request.', true)
+    ->param('offset', 0, new Range(0, 2000), 'Results offset. The default value is 0. Use this param to manage pagination.', true)
+    ->param('cursor', '', new UID(), 'ID of the bucket used as the starting point for the query, excluding the bucket itself. Should be used for efficient pagination when working with large sets of data.', true)
+    ->param('cursorDirection', Database::CURSOR_AFTER, new WhiteList([Database::CURSOR_AFTER, Database::CURSOR_BEFORE]), 'Direction of the cursor.', true)
+    ->param('orderType', 'ASC', new WhiteList(['ASC', 'DESC'], true), 'Order result by ASC or DESC order.', true)
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->inject('usage')
+    ->action(function ($search, $limit, $offset, $cursor, $cursorDirection, $orderType, $response, $dbForInternal, $usage) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+        /** @var Appwrite\Stats\Stats $usage */
+
+        $queries = ($search) ? [new Query('name', Query::TYPE_SEARCH, $search)] : [];
+        
+        if (!empty($cursor)) {
+            $cursorBucket = $dbForInternal->getDocument('buckets', $cursor);
+
+            if ($cursorBucket->isEmpty()) {
+                throw new Exception("Bucket '{$cursor}' for the 'cursor' value not found.", 400);
+            }
+        }
+
+        $usage->setParam('storage.buckets.read', 1);
+
+        $response->dynamic(new Document([
+            'buckets' => $dbForInternal->find('buckets', $queries, $limit, $offset, [], [$orderType], $cursorBucket ?? null, $cursorDirection),
```

Comment:
What attribute are we ordering by?

Reply:
ya right? Thinking the same I just copied the same behavior from collection here https://github.com/appwrite/appwrite/blob/252720dd2874e42988fff8f8c3619b0eeeebb0fa/app/controllers/api/database.php#L234
?